scripts/upgrade.sh

```diff
@@ -0,0 +1,243 @@
+#!/bin/bash
+
+#Author: Scality MetalK8s
+#
+#Summary: This script is used to upgrade Metalk8s cluster and all the components
+#          that make up the stack.
+#
+set -e
+set -u
+# The trickery below is to only run `set -o pipefail`, which is a Bash'ism,
+# when running in Bash and not some other `sh` which doesn't necessarily
+# support `-o pipefail` (though must support `-e` and `-u` per POSIX).
+if test -z "$(type -p)"; then set -o pipefail; fi
+
+# set -o nounset # Treat unset variables as an error
+
+#ANSI color codes:
+RED='\033[1;31m'
+GREEN='\033[1;32m'
+NOCOLOR='\033[0m'
+
+SALT_CALL=${SALT_CALL:-salt-call}
+
+VERBOSE=${VERBOSE:-0}
+LOGFILE=/var/log/metalk8s-upgrade.log
+
+# Some helpful guide on upgrade options
+__usage() {
+    cat << EOF
+  Usage :  ${__ScriptName} <options>
+  Options:
+    -h | --help              Display HELP options
+    -v | --version           Show current installed version
+    -d | --dry-run           Show DRYRUN output.
+    -t | --to                Upgrade to specified version
+         --verbose           Run in verbose mode
+EOF
+}
+
+
+TMPFILES=$(mktemp -d)
+
+cat << EOF >> "${LOGFILE}"
+--- MetalK8s Upgrade started on $(date -u -R) ---
+EOF
+
+exec > >(tee -ia "${LOGFILE}") 2>&1
+
+cleanup() {
+    rm -rf "${TMPFILES}" || true
+}
+
+trap cleanup EXIT
+
+
+run_quiet() {
```

Comment:
It seems to me that we have a lot of copy/paste between our scripts.

We may want (not in this PR, it's out of scope) to have a shell script library with our common function inside, no?

Reply:
yes I agree

Reply:
Tracked in #1441